### PR TITLE
Don't draw Level Up Icon if quest log opened

### PR DESCRIPTION
**Pull request by de4me**
Снимок экрана 2019-11-08 в 10 13 25

### CODE REVIEW

**Comment by AJenbo** [comment]
This check was changed to better support widescreen (where the panel doesn't conflict with with the icon. Could I get you to change it to also check if there is room for it on either height (SPANEL_HEIGHT+PANEL_HEIGHT+49) or width (4*320).

**Comment by de4me** [comment]
Wide screen, I can run only in 640x480 with scaling)
Ok. I will try to find out how wide format support is turned on.

**Comment by AJenbo** [comment]
You don't really need to turn it on to do the math :)

To enable change the resolution you currently have to edit SCREEN_WIDTH and SCREEN_HEIGHT in defs.h, but once everything look good this will become doable at boottime, and eventually even runtime.

File: Source/scrollrt.cpp [hunk 1]
```diff
@@ -892,8 +892,7 @@ void DrawView(int StartX, int StartY)
 		DrawChr();
 	} else if (questlog) {
 		DrawQuestLog();
-	}
-	if (!chrflag && plr[myplr]._pStatPts != 0 && !spselflag) {
+	} else if (plr[myplr]._pStatPts != 0 && !spselflag) {
```
Comment (AJenbo) [inline Source/scrollrt.cpp]: This should work

```suggestion
	}
	if (!chrflag && plr[myplr]._pStatPts != 0 && !spselflag
		&& (!questlog || SCREEN_HEIGHT >= SPANEL_HEIGHT + PANEL_HEIGHT + 49 || SCREEN_WIDTH >= 4*320)) {
```

**Comment by de4me** [comment]
I play with HEIGHT and WIDTH, but don't start the game.
Only game screen support wide format?

Снимок экрана 2019-11-08 в 13 58 51

**Comment by AJenbo** [comment]
Yes, only the basics are done atm.

**Comment by de4me** [comment]
What if i move questlog back to panel on wide screen?
I think it’s better for UI if all the elements are together.

**Comment by AJenbo** [comment]
No we prefer it to be in the upper corner

File: Source/scrollrt.cpp [hunk 2]
```diff
@@ -893,7 +893,8 @@ void DrawView(int StartX, int StartY)
 	} else if (questlog) {
 		DrawQuestLog();
 	}
-	if (!chrflag && plr[myplr]._pStatPts != 0 && !spselflag) {
+	if (!chrflag && plr[myplr]._pStatPts != 0 && !spselflag
+		&& (!questlog || SCREEN_HEIGHT >= SPANEL_HEIGHT + PANEL_HEIGHT + 49 || SCREEN_WIDTH >= 4 * 320)) {
```
Comment (qndel) [inline Source/scrollrt.cpp]: what's this

Comment (qndel) [reply Source/scrollrt.cpp]: why 4* 320 instead of 1280? :D

Comment (AJenbo) [reply Source/scrollrt.cpp]: Space for 4 * panels. 1280 is a magic number with no meaning :P

Comment (MaxDesiatov) [reply Source/scrollrt.cpp]: `4 * PANEL_WIDTH` would make more sense then 🙂

Comment (qndel) [reply Source/scrollrt.cpp]: sadly PANEL_WIDTH is 640 as it's the main panel's width - there's no define for 320 - side panel width :D

Comment (AJenbo) [reply Source/scrollrt.cpp]: It might make sens to add `SPANEL_WIDTH 320`

**Comment by de4me** [comment]
Снимок экрана 2019-11-08 в 20 56 23
Снимок экрана 2019-11-08 в 21 02 02

File: Source/scrollrt.cpp [hunk 3]
```diff
@@ -893,8 +893,11 @@ void DrawView(int StartX, int StartY)
 	} else if (questlog) {
 		DrawQuestLog();
 	}
-	if (!chrflag && plr[myplr]._pStatPts != 0 && !spselflag
-		&& (!questlog || SCREEN_HEIGHT >= SPANEL_HEIGHT + PANEL_HEIGHT + 49 || SCREEN_WIDTH >= 4 * 320)) {
+#if SCREEN_HEIGHT-PANEL_HEIGHT-SPANEL_HEIGHT >= 74 || PANEL_X > SPANEL_WIDTH
```
Comment (AJenbo) [inline Source/scrollrt.cpp]: Pre processor statements will make it harder to implement run time resolutions as they will have to be rewritten. Also you want PANEL_LEFT as _X is in relation to the back buffer and not the screen

Comment (de4me) [reply Source/scrollrt.cpp]: But this is optimized way. Performance depends on this procedure.
When you make resolution switching support, you change this code.

Comment (AJenbo) [reply Source/scrollrt.cpp]: Performance optimizing when there isn't a performance issue is known as premature optimization. It leads to a complicated code that is hard to maintain. It isn't too long ago that I cleaned up a lot of this from the original code https://github.com/diasurgical/devilutionX/pull/361. I don't plan to merge code that adds more code that we know will have to be changed any way. Please make the requested changes for this PR to be ready for merge.

Comment (AJenbo) [reply Source/scrollrt.cpp]: P.s. the compiler would optimize away a comparison of static numbers so there actually din't any performance difference between the two solutions.

Comment (de4me) [reply Source/scrollrt.cpp]: I understand you, but I think differently.
I close the request, as I see all my commits are displayed at you and you can merge them.

Comment (AJenbo) [reply Source/scrollrt.cpp]: Ok, but we will only accept PRs that follow the project standard. The simple reason is that there isn't much point in swapping one issue for another.

**Comment by AJenbo** [comment]
Fixed here https://github.com/diasurgical/devilutionX/commit/34a6615fedb87247fc6ed47ca68ba7a75cb3327e

**Pull request closed without merging**